src/replacement/simple_replace.rs

```diff
@@ -0,0 +1,228 @@
+use std::collections::{HashMap, HashSet};
+
+use crate::{hugr::Node, Hugr, Port};
+use thiserror::Error;
+
+/// Specification of a simple replacement operation.
+#[derive(Debug, Clone)]
+pub struct SimpleReplacement {
+    /// The common DFG parent of all nodes to be replaced.
+    pub p: Node,
+    /// The set of nodes to remove (a convex set of leaf children of `p`).
```

Comment:
How about calling this `remove` (with comment "called S in the spec" if you like!)

Reply:
I propose renaming:

* `p` --> `region`
* `s` --> `removal`
* `n` --> `replacement`

Reply:
I love `removal` and agree with `replacement`. Less sure about `region` - maybe just `parent`?

Reply:
Changed to `parent`.